README.md

```diff
@@ -10,36 +9,139 @@ license](APACHE-2.0.txt) and documentation is provided under the [CC-BY-4.0
+specify the base configuration template as well as individual configuration items. The common
+usage will be to specify a template for the general environment (e.g. cql-es) and then provide
+additional individual configuration to override/extend the template.
+
+The `JANUS_PROPS_TEMPLATE` environment variable is used to specify the JanusGraph configuration
+template. Properties in the template are used unless overridden individually as described below.
+The available templates depend on the JanusGraph version.
+
+| JANUS_PROPS_TEMPLATE | Supported Versions |
+| -------------------- | ------------------ |
+| berkeleyje (default) | all                |
+| berkeleyje-es        | all                |
+| cassandra-es         | all                |
+| cql-es               | >=0.2.1            |
+
+Individual JanusGraph configuration items can be configured using corresponding environment
+variables. Provided configuration items will be used to override/extend the template. See
+[JanusGraph Configuration][JG_CONFIG] for more information on available configuration options.
+Gremlin server configuration items can also be specified in the environment by using the
+`server` namespace.
 
 **Example**
 
-| Configuration Key       | Environment Variable        |
-| ----------------------- | --------------------------- |
-| storage.backend         | JANUS_STORAGE_BACKEND       |
-| storage.cql.keyspace    | JANUS_STORAGE_CQL_KEYSPACE  |
-| index.search.backend    | JANUS_INDEX_SEARCH_BACKEND  |
+Start a JanusGraph instance with custom storage and server settings
+
+```bash
+$ docker run --name janusgraph-default -e storage.berkeleyje.cache-percentage=80 -e server.threadPoolWorker=2 janusgraph/janusgraph:latest
+```
 
+Inspect configuration
 
-In general, a configuration parameter of the form `chained.property.path` will take the form
-`JANUS_CHAINED_PROPERTY_PATH`. The environment variable will be prefixed with `JANUS` and the 
-hyphens and periods in the name of the property key are replaced with underscores. Lastly, the
-characters in the property path are converted to uppercase.
+```bash
+$ docker exec janusgraph-default sh -c 'cat /etc/opt/janusgraph/janusgraph.properties | grep ^[a-z]'
+gremlin.graph=org.janusgraph.core.JanusGraphFactory
+storage.backend=berkeleyje
+storage.directory=/var/lib/janusgraph/data
+index.search.backend=lucene
+storage.berkeleyje.cache-percentage=80
+index.search.directory=/var/lib/janusgraph/index
+$ docker exec janusgraph-default grep threadPoolWorker /etc/opt/janusgraph/gremlin-server.yaml
+threadPoolWorker: 2
+```
+
+**Example**
+
+Start a JanusGraph instance with Cassandra and Elasticsearch using the `cql-es` template through
+[docker-compose-cql-es.yml](docker-compose-cql-es.yml)
+
+```bash
+$ docker-compose -f docker-compose-cql-es.yml up
+```
+
+Note that the `index.search.backend` property is set to `lucene` in the default environment.
```

Comment:
Can't we solve that by using `berkeleyje-lucene` as the default template instead of `berkeleyje`? Then we wouldn't have to set `index.search.backend=lucene` in the `Dockerfile`.

Reply:
Agreed, definitely don't like this caveat. Currently it's pulling templates from conf/gremlin-server and I don't see one in there for the berkeleyje-lucene case (unfortunately).

https://github.com/search?q=org:JanusGraph+repo:janusgraph+filename:janusgraph.properties%20path:janusgraph-dist/src/assembly/static/conf/gremlin-server

Reply:
Ok, that's bad. I only looked here:

https://github.com/JanusGraph/janusgraph/blob/master/janusgraph-dist/src/assembly/cfilter/conf/janusgraph-berkeleyje-lucene.properties

which contains a lot more of these `.properties` files, but in a template form.

Reply:
Ah, I see, you create this file now yourself in the Dockerfile 👍